jobrunner/cli/local_run.py

```diff
@@ -478,13 +469,25 @@ def filter_log_messages(record):
     Not all log messages are useful in the local run context so to avoid noise
     and make things clearer for the user we filter them out here
     """
-    status_code = getattr(record, "status_code", None)
-    if status_code in STATUS_CODES_NOT_TO_LOG:
+    # None of these status messages are particularly useful in local run
+    # mode, and they can generate a lot of clutter in large dependency
+    # trees
+    if getattr(record, "status_code", None) in {
+        StatusCode.WAITING_ON_DEPENDENCIES,
+        StatusCode.DEPENDENCY_FAILED,
+        StatusCode.WAITING_ON_WORKERS,
+    }:
         return False
+
     # We sometimes log caught exceptions for debugging purposes in production,
     # but we don't want to show these to the user when running locally
     if getattr(record, "exc_info", None):
         return False
+
+    # Executor state logging is pretty verbose and unlikely to be useful for local running
+    if record.name == executors.logging.LOGGER_NAME:
+        return False
```

Comment:
Nice, this is actually quite clean in the end I think?

Reply:
Yes, especially since we already filter the logs for local running, which I didn't realize before I started. And _especially_ especially because @evansd refactored that stuff while I was away to make it cleaner.